docs/sql-migration-guide-upgrade.md

```diff
@@ -50,6 +50,8 @@ license: |
 
   - In Spark version 2.4 and earlier, JSON datasource and JSON functions like `from_json` convert a bad JSON record to a row with all `null`s in the PERMISSIVE mode when specified schema is `StructType`. Since Spark 3.0, the returned row can contain non-`null` fields if some of JSON column values were parsed and converted to desired types successfully.
 
+  - In Spark version 2.4 and earlier, if data columns overlap with partition columns, the output schema of file scan respects the ordering of data columns, and adopts the data type of partition columns. Since Spark 3.0, the output schema of file scan puts all the partition columns at the end. For example, if the data schema is `[a: String, b: String, c: String]` and the partition schema is `[b: Int, d: Int]`, the result schema is `[a: String, b: Int, c: String, d: Int]` in Spark 2.4 and earlier, and `[a: String, c: String, b: Int, d: Int]` since Spark 3.0.
```

Comment:
do we need migration guide? it's a behavior change for file source v2, which is new in Spark 3.0.

Reply:
I am OK with either way. Let me remove this.